src/Provider/TaxRatePercentageProvider.php

```diff
@@ -34,7 +34,10 @@ public function provideFromAdjustable(AdjustableInterface $adjustable): ?string
             return null;
         }
 
-        return $this->provideFromAdjustment($taxAdjustments->first());
+        $adjustment = $taxAdjustments->first();
```

Comment:
I can't mark more lines as they're out of range, but you can do something like that:

Remove the `if` with `isEmpty()` and write it like this:

```suggestion
        $adjustment = $taxAdjustments->first();
        
        if (null === $adjustment) {
            return null;
        }
        
        return $this->provideFromAdjustment($adjustment);
```

Reply:
I think both solutions are equivalent as we need to check there is a single adjustment. The previous solution was getting this information based on collection API, now we check it directly from the returned collection element

Reply:
I know, but after my suggestion `Assert::isInstanceOf($adjustment,AdjustmentInterface::class);` can be removed, and it should work I guess 🤔.

Reply:
We've solved it together 🤝🏼.